stdlib/builtin.rbi

```diff
@@ -560,10 +560,60 @@ class Enumerator<'a, 'b>
                 | -> Enumerator<['a, Integer], 'b>
 end
 
+class Encoding
+end
+
 class Regexp
   def self.compile: (String, *any) -> Regexp
+                  | (Regexp) -> Regexp
   def self.escape: (String) -> String
+  def self.last_match: -> MatchData?
+                     | (Integer) -> String?
+  def self.quote: (String) -> String
+  def self.try_convert: (any) -> Regexp?
+  def self.union: (*String) -> Regexp
+                | (Array<String>) -> Regexp
+                | (*Regexp) -> Regexp
+                | (Array<Regexp>) -> Regexp
+
+  def initialize: (String, *any) -> any
```

Comment:
Perhaps we may be able to use more strict type instead of `*any`, but I don't know how to define such a type... :sweat:

```diff
-def initialize: (String, *any) -> any
+def initialize: (String) -> any
               | (String, Integer) -> any
               | (String, **non_nil_or_false**) -> any
```

API Doc says:

> If options is an Integer, it should be one or more of the constants Regexp::EXTENDED, Regexp::IGNORECASE, and Regexp::MULTILINE, or-ed together. Otherwise, if options is not nil or false, the regexp will be case insensitive.

http://ruby-doc.org/core-2.5.1/Regexp.html#method-c-new

(Also `Regexp.compile`)

Reply:
What we want exactly should be:

```
def initialize: (String) -> any           # no extra parameter
              | (String, Integer) -> any  # for Regexp options are given
              | (String, bool) -> any     # for any other values
```

Reply:
I agree with your idea. 👍